packages/components/src/mobile/image/index.native.js

```diff
@@ -259,14 +289,57 @@ const ImageComponent = ( {
 					</View>
 				) : (
 					<View style={ focalPoint && styles.focalPointContent }>
-						<Image
-							style={ imageStyles }
-							source={ { uri: url } }
-							{ ...( ! focalPoint && {
-								resizeMethod: 'scale',
-							} ) }
-							resizeMode={ imageResizeMode }
-						/>
+						{ Platform.isAndroid && (
+							<>
+								{ networkImageLoaded && networkURL && (
+									<Animated.Image
+										style={ imageStyles }
+										fadeDuration={ 0 }
+										source={ { uri: networkURL } }
+										{ ...( ! focalPoint && {
+											resizeMethod: 'scale',
+										} ) }
+										resizeMode={ imageResizeMode }
+									/>
+								) }
+								{ ! networkImageLoaded && ! networkURL && (
+									<Animated.Image
+										style={ imageStyles }
+										fadeDuration={ 0 }
+										source={ { uri: localURL } }
+										{ ...( ! focalPoint && {
+											resizeMethod: 'scale',
+										} ) }
+										resizeMode={ imageResizeMode }
+									/>
+								) }
+							</>
+						) }
+						{ Platform.isIOS && (
+							<>
+								<Animated.Image
+									style={ imageStyles }
+									source={ {
+										uri:
+											networkURL && networkImageLoaded
+												? networkURL
+												: ( localURL && localURL ) ||
+												  url,
+									} }
+									{ ...( ! focalPoint && {
+										resizeMethod: 'scale',
+									} ) }
+									resizeMode={ imageResizeMode }
+								/>
+								<Image
+									source={ { uri: networkURL } }
+									style={ styles.nonVisibleImage }
+									onLoad={ () => {
+										setNetworkImageLoaded( true );
```

Comment:
I was trying to consider if `networkImageLoaded` could be removed entirely as it seems a bit redundant, but I suppose it is necessary for the iOS implementation specifically. Correct?

Reply:
Correct. On iOS, `networkURL` alone cannot be used as a boolean to trigger that the network image is ready to be displayed. There is a period of time where `networkURL` is known as a string that _could_ be used as a boolean (`https://site.com/network-image.jpg`), but the image data itself has not been downloaded. In the current iOS implementation, the brief period of time where this occurs is the source of the disappearing image.

In essence, the lack of a callback from the FastImage [`preload`](https://github.com/DylanVann/react-native-fast-image/?tab=readme-ov-file#fastimagepreload-source--void) method is the root of the entire problem on iOS, and the reason for the divergence between the two platforms. My perception is that FastImage's preload method was simply not designed for our use case of setting local image data and replacing it with network image data -- rather, it seems designed for use with static placeholders to be replaced by network image data as soon as it's available.